symfit/core/objectives.py

```diff
@@ -0,0 +1,216 @@
+        evaluated_func = self.model(**self.independent_data, **parameters)
+        result = [0.0 for _ in self.model.params]
+
+    def eval_jacobian(self, **parameters):
+        chi = self(flatten=False, **parameters)
+        jac_kwargs = dict(**self.independent_data, **parameters)
+        evaluated_func = self.model(**self.independent_data, **parameters)
+
+        result = len(self.model.params) * [0.0]
+        for ans, y, row in zip(evaluated_func, self.model, self.model.numerical_jacobian):
+            if self.dependent_data[y.name] is not None:
+                for index, component in enumerate(row):
+                    result[index] += component(**jac_kwargs) * (
+                        (self.dependent_data[y.name] - ans) / self.sigma_data[self.model.sigmas[y].name] ** 2
+                    )
+        result *= (1 / chi)
+        result = np.nan_to_num(result)
+        result = [item.flatten() for item in result]
+        return - np.array(result).T
+
+
+class LeastSquares(GradientObjective):
+    @keywordonly(flatten_components=True)
+    def __call__(self, **parameters):
+        flatten_components = parameters.pop('flatten_components')
+        jac_kwargs = dict(**self.independent_data, **parameters)
+        evaluated_func = self.model(**jac_kwargs)
+
+        chi2 = [0 for _ in evaluated_func]
+        for index, (dep_var_name, dep_var_value) in enumerate(evaluated_func._asdict().items()):
+            dep_data = self.dependent_data[dep_var_name]
+            if dep_data is not None:
+                sigma = self.sigma_data['sigma_{}'.format(dep_var_name)]  # Should be changed with #41
+                chi2[index] += np.sum(
+                    (dep_var_value - dep_data) ** 2 / sigma ** 2)
+                # chi2 += np.sum((dep_var_value - dep_data)**2/sigma**2)
+        chi2 = np.sum(chi2) if flatten_components else chi2
+        return chi2
+
+    def eval_jacobian(self, **parameters):
+        # jac_args = list(independent_data.values()) + list(p)
+        evaluated_func = self.model(**self.independent_data, **parameters)
+        result = [0.0 for _ in self.model.params]
+
+        for ans, var, row in zip(evaluated_func, self.model,
+                               self.model.numerical_jacobian):
+            dep_data = self.dependent_data[var.name]
+            sigma_var = self.model.sigmas[var]
+            if dep_data is not None:
+                sigma = self.sigma_data[sigma_var.name]  # Should be changed with #41
+                for index, component in enumerate(row):
+                    result[index] += np.sum(
+                        component(**self.independent_data, **parameters) * ((dep_data - ans) / sigma ** 2)
+                    )
+        return - np.array(result).T
+
+
+class LogLikelihood(GradientObjective):
+    def __call__(self, **parameters):
+        """
+        Error function to be maximised(!) in the case of log-likelihood fitting.
```

Comment:
Doesn't it make more sense to agree that all objectives should *always* be minimized, and flip the sign here?

Reply:
Not sure I understand this comment, because the sign is indeed flipped by default here.
`ans = - np.nansum(np.log(self.model(**jac_kwargs)))`

This is more intuitive I think in this case. Or is your mark about the (exclamation point in the) docstring?

Reply:
In that case the docstring lies, because it says it should be maximised.